lesson05/circle_test.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"math"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAreaC(t *testing.T) {
+
+	tests := []struct {
+		name    string
+		args    Circle
+		want    float64
+		wantErr bool
+	}{
+		{
+			name: "if radius is <= 0 should return error ",
+			args: Circle{
+				Radius: 0.0,
+			},
+			want:    0.0,
+			wantErr: true,
+		},
+		{
+			name: "if radius = 1 should return Pi",
+			args: Circle{
+				Radius: 1.0,
+			},
+			want:    math.Pi,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := (tt.args).Area()
+
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Area() error = %v, wantErr %v, got = %v", err, tt.wantErr, got)
+				assert.Equal(t, err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
```

Comment:
`reflect.DeepEqual` is usually used to compare structs, that have nested structure in them, basically other structs as fields, and other complex types, like maps and slices. 
Comparing floats in any language is not going to work with `==` sign, or at least not always, due to the way they are represented in binary. Right now this assertion is never reached, because the statement `!reflect.DeepEqual(got, tt.want)` is false.
There is a way to compare floats with a certain precision tolerance, however. One way to do that is shown [here](https://golangbyexample.com/comparing-floating-point-numbers-go/), for example, in the section "Comparing Same Float Types With Tolerance"